### ISSUE TITLE
quantization.fuse_modules fails with Conv1d and BatchNorm1d

### ISSUE THREAD

**Issue by lamipaul** [pytorch/pytorch#41534]
## 🐛 Bug

A `NotImplementedError` is raised when trying to fuse a Conv1d and a BatchNorm1d using the` torch.quantization.fuse_modules` function

## To Reproduce

```
import torch
m = torch.nn.Sequential(torch.nn.Conv1d(1,1,3), torch.nn.BatchNorm1d(1))
torch.quantization.fuse_modules(m, [['0','1']])
```

## Expected behavior
The function should work with conv and bn, but raises a NotImplementedError.
the keys needed in [the source](https://github.com/pytorch/pytorch/blob/d80e0c62bea9cc4f0b3c46edbcf892baee38b4c8/torch/quantization/fuse_modules.py#L106) are` torch.nn.Conv1d`, and `torch.nn.BatchNorm1d`, when the actual module types are `torch.nn.modules.conv.Conv1d` and `torch.nn.modules.batchnorm.BatchNorm1d`

## Environment
PyTorch version: 1.5.1
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.8
Is CUDA available: No
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: TITAN X (Pascal)
GPU 1: TITAN X (Pascal)

Nvidia driver version: 418.67
cuDNN version: Probably one of the following:
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.1.10
/usr/local/cuda-9.1/targets/x86_64-linux/lib/libcudnn.so.7.1.3

Versions of relevant libraries:
[pip3] numpy==1.18.1
[pip3] torch==1.5.1
[pip3] torchaudio==0.4.0a0+ba6d67b
[pip3] torchfile==0.1.0
[pip3] torchvision==0.5.0
[conda] _pytorch_select           0.2                       gpu_0  
[conda] blas                      1.0                         mkl  
[conda] cuda75                    1.0                  hf2493ae_0    pytorch
[conda] cudatoolkit               10.0.130                      0  
[conda] mkl                       2020.0                      166  
[conda] mkl-service               2.3.0            py38he904b0f_0  
[conda] mkl_fft                   1.0.15           py38ha843d7b_0  
[conda] mkl_random                1.1.0            py38h962f231_0  
[conda] numpy                     1.18.1           py38h4f9e942_0  
[conda] numpy-base                1.18.1           py38hde5b4d6_1  
[conda] torch                     1.5.1                    pypi_0    pypi
[conda] torchaudio                0.4.0                      py38    pytorch
[conda] torchelie                 0.1.dev0                 pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchvision               0.5.0                    pypi_0    pypi

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a @vkuzo

**Comment by vkuzo** [comment]
hi @lamipaul , thanks for the report.  Note: you can already do this if the model is in eval mode.  To support QAT, we'll need to add `torch.nn.intrinsic.qat.ConvBn1d`.

**Comment by naveenggmu** [comment]
@lamipaul Were you able to solve this issue? I am not able to fuse conv1d and batchnorm1d as well. Getting the same error.

**Comment by lamipaul** [comment]
Hi @naveenggmu and @vkuzo 
Having the model in eval mode indeed solves the problem, at least for pytorch 1.6 (not for 1.4)

Thanks

**Comment by lamipaul** [comment]
Similarly to the mentioned problem with Conv and BN, the problem persists with activation functions such as LeakyReLU

`import torch
m = torch.nn.Sequential(torch.nn.Conv1d(1,1,3), torch.nn.BatchNorm1d(1), torch.nn.LeakyReLU())
m.eval()
torch.quantization.fuse_modules(m, [['0','1', '2']])
`
yields the same error

**Comment by vkuzo** [comment]
Here (https://github.com/pytorch/pytorch/blob/0c77bd7c0bbd4d6e50a5f3ce7b4debbee85d7963/torch/quantization/fuse_modules.py#L106) is a list of modules which can be fused.  LeakyRELU fusion does not work because we don't have a fusion implemented for it.

**Comment by vkuzo** [comment]
removing v1.7 since the request was for static quantization, which works if the model is in eval mode.
